connector-pagerduty/src/main/java/com/redhat/cloud/notifications/connector/pagerduty/PagerDutyRouteBuilder.java

```diff
@@ -0,0 +1,74 @@
+package com.redhat.cloud.notifications.connector.pagerduty;
+
+import com.redhat.cloud.notifications.connector.EngineToConnectorRouteBuilder;
+import com.redhat.cloud.notifications.connector.authentication.secrets.SecretsLoader;
+import com.redhat.cloud.notifications.connector.http.HttpConnectorConfig;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+
+import static com.redhat.cloud.notifications.connector.ConnectorToEngineRouteBuilder.SUCCESS;
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.ID;
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.ORG_ID;
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.TARGET_URL;
+import static com.redhat.cloud.notifications.connector.http.SslTrustAllManager.getSslContextParameters;
+import static com.redhat.cloud.notifications.connector.pagerduty.ExchangeProperty.ACCOUNT_ID;
+import static com.redhat.cloud.notifications.connector.pagerduty.ExchangeProperty.TARGET_URL_NO_SCHEME;
+import static com.redhat.cloud.notifications.connector.pagerduty.ExchangeProperty.TRUST_ALL;
+import static org.apache.camel.Exchange.CONTENT_TYPE;
+import static org.apache.camel.LoggingLevel.INFO;
+import static org.apache.camel.builder.endpoint.dsl.HttpEndpointBuilderFactory.HttpEndpointBuilder;
+
+// TODO: rewrite for PagerDuty
+@ApplicationScoped
+public class PagerDutyRouteBuilder extends EngineToConnectorRouteBuilder {
+
+    private static final String APPLICATION_JSON = "application/json; charset=utf-8";
+
+    static final String PAGERDUTY_RESPONSE_TIME_METRIC = "micrometer:timer:pagerduty.response.time";
+    static final String TIMER_ACTION_START = "?action=start";
+    static final String TIMER_ACTION_STOP = "?action=stop";
+
+    @Inject
+    HttpConnectorConfig connectorConfig;
+
+    @Inject
+    SecretsLoader secretsLoader;
+
+    @Inject
+    AuthenticationProcessor authenticationProcessor;
+
+    @Override
+    public void configureRoutes() {
+        from(seda(ENGINE_TO_CONNECTOR))
+            .setHeader(CONTENT_TYPE, constant(APPLICATION_JSON))
+            .routeId(connectorConfig.getConnectorName())
+            .process(secretsLoader)
+            .process(authenticationProcessor)
+            .to(PAGERDUTY_RESPONSE_TIME_METRIC + TIMER_ACTION_START)
+                // SSL certificates may or may not be verified depending on the integration settings.
+                .choice()
+                .when(exchangeProperty(TRUST_ALL))
+                    .toD(buildPagerDutyEndpoint(true), connectorConfig.getEndpointCacheMaxSize())
+                .endChoice()
+                // TODO is this a valid option for PagerDuty?
+                .otherwise()
+                    .toD(buildPagerDutyEndpoint(false), connectorConfig.getEndpointCacheMaxSize())
+                .end()
```

Comment:
Maybe we could ask AppSre to check if they think that having PagerDuty installations on an HTTP endpoint is common, or if we should simply force everyone to use HTTPS for the integration :thinking: Can you please verify that?

Reply:
From my research, PagerDuty isn't available as a self-hosted option, which means that all requests must go to <https://events.pagerduty.com/v2/enqueue>. I can still follow up with them, but I think we can just drop it entirely.

This applies to a lot of the variables for PagerDuty, which I mentioned a bit in the **Payload schema** section. I've started working on that in [the other branch](https://github.com/cam-rod/notifications-backend/blob/fcc0ef6507771699a9cb290bebcc7675ebc314af/common/src/main/java/com/redhat/cloud/notifications/models/PagerDutyProperties.java#L27-L37) if you're curious.

Reply:
Perfect, thank you for checking that out! Then yes, let's remove that branch.

Reply:
Removed, and clarifying that the branch kept is `.toD(buildPagerDutyEndpoint(false), connectorConfig.getEndpointCacheMaxSize())` since it doesn't disable TLS verification